scalr/data_source_scalr_agent_pool.go

```diff
@@ -45,17 +46,26 @@ func dataSourceScalrAgentPool() *schema.Resource {
 
 func dataSourceScalrAgentPoolRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	scalrClient := meta.(*scalr.Client)
-	var envID string
 
+	agentPoolID := d.Get("id").(string)
 	name := d.Get("name").(string)
 	accountID := d.Get("account_id").(string)
+	envID := d.Get("environment_id").(string)
+
 	options := scalr.AgentPoolListOptions{
-		Name:    name,
```

Comment:
Now that the `name` is not required the following configuration gives us the first agent pool in the account, which is kind of weird:
```
data "scalr_agent_pool" "pool" {
  id   = ""
  name = ""
  # one of the id, name can be omitted with the same result
}

output o {
  value = data.scalr_agent_pool.pool
}
```
⬇️
```
GET /agent-pools?filter[account]=acc-***
```
⬇️
```
Changes to Outputs:
  + o = {
      + account_id     = "acc-***"
      + environment_id = null
      + id             = "apool-***"
      + name           = "my-pool"
      + workspace_ids  = null
    }
```

`AtLeastOneOf` loses sense in this case, as the configuration above would be equivalent (in terms of result) to just
```
data "scalr_agent_pool" "pool" {}
```

I'd suggest adding a `StringIsNotEmpty` or `StringIsNotWhiteSpace` validation function for such arguments, hence restricting empty values declaratively at early stage.

Need to check it across all affected datasources.

Reply:
Added `StringIsNotWhiteSpace` validation function.